### ISSUE TITLE
Do not navigate to the default route when a route doesn't match (broken links on the KNB tools page)

### ISSUE THREAD

**Issue by amoeba** [NCEAS/metacatui#787]
Just noticed that a few of the links on the KNB tools page, such as https://knb.ecoinformatics.org/software/dist/MorphoUserGuide.pdf don't work if left-clicked but do it if I open in a new tab. It's pretty obvious what's going on but I'm not totally sure what to do here.

It looks like the current application logic is to render the default route if the requested route isn't in the routes table. Makes sense. Now that we've refactored the routing, we need a way to differentiate between route non-matches that should 404 and those that shouldn't.

A way around deciding this would be to add `target` attributes on the anchor elements to URLs that'd result in a route miss. Thoughts?

**Comment by amoeba** [comment]
Chatted with @csjx about this today after @taojing2002 	and he noticed a related issue. The overarching issue is that it's a bit wonky to have a webapp that doesn't control its entire URL space. On some deploys, we mix MetacatUI and Metacat's URLs together and we don't make use of a mapping to differentiate them.

It appears that we could extend the click handler (hijacker) in `app.js` to be aware of URLs in Metacat's URL space via the deployment context configuration value:

https://github.com/NCEAS/metacatui/blob/7bf7dc363e5bed5b51ecdef441ac1da922effee3/src/js/app.js#L256-L258

Basically we could just change the test to include `href.attr.indexOf(context) > 0` where `context` would come out of the config.

Thoughts?

**Comment by amoeba** [comment]
PS: The above only addresses _part_ of the broken link issue. Sorry I forgot to note that!

**Comment by laurenwalker** [comment]
We could add a configuration to the AppModel called something like `ignorePaths` that the click handler will ignore. They would be paths relative to the MetacatUI.root. In the KNB case, the AppModel would be configured like this:

` ignorePaths: ["/software", "/knb"] `

**Comment by laurenwalker** [comment]
We could also remove the functionality where a default route is used, and instead if a route doesn't match, just route the person to the given URL. If there is a 404, they will end up back at MetacatUI anyway because of the `FallbackResource`. I think I like this option the best.

**Comment by amoeba** [comment]
Oh yeah! That actually works now since we use FallbackResource? I think I like that more.

**Comment by laurenwalker** [comment]
Cool, I'm going to rename this issue since I think that is the way to go
